cmd/object-handlers.go

```diff
@@ -1331,6 +1363,14 @@ func (api objectAPIHandlers) DeleteObjectHandler(w http.ResponseWriter, r *http.
 		return
 	}
 
+	// Deny if WORM is enabled
+	if globalWORMEnabled {
```

Comment:
We should perform WORM related check *only after* determining if the request is correctly signed *or* anonymous request is allowed via bucket policy.

Reply:
Done